File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.cpp
class Solution {
public:
        int longestOnes(vector<int>& A, int K) {
        vector<int> freq(2);
        int i = 0, r=0;
        for(int j = 0; j < A.size(); j++){
            freq[A[j]] ++;
            while(freq[0] > K && i <= j){
                freq[A[i]] --;
                i++;
            }
            r = max(r, j-i+1);
        }
        return r;
    }
};

File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        c=0
        ans=0
        i=0
        for j in range(len(nums)):
            if nums[j]==0:
                c+=1
            if c>k:
                if nums[i]==0:
                    c-=1
                i+=1
            ans=max(ans,j-i+1)
        return ans
